src/test/java/org/apache/commons/lang3/reflect/TypeUtilsTest.java

```diff
@@ -539,360 +521,243 @@ public void testIsArrayTypeClasses() {
-        assertTrue(TypeUtils.isAssignable(int.class, intComparableType));
         assertTrue(TypeUtils.isAssignable(int.class, Comparable.class));
-        final Serializable ser = 1;
         assertTrue(TypeUtils.isAssignable(int.class, Serializable.class));
-        final Type longComparableType = getClass().getField("longComparable").getGenericType();
-        // longComparable = 1;
-        assertFalse(TypeUtils.isAssignable(int.class, longComparableType));
-        // longComparable = Integer.valueOf( 0 );
-        assertFalse(TypeUtils.isAssignable(Integer.class, longComparableType));
-        // int[] ia;
-        // long[] la = ia;
+
         assertFalse(TypeUtils.isAssignable(int[].class, long[].class));
-        final Integer[] ia = null;
-        final Type caType = getClass().getField("intWildcardComparable").getGenericType();
-        intWildcardComparable = ia;
-        assertTrue(TypeUtils.isAssignable(Integer[].class, caType));
-        // int[] ina = ia;
         assertFalse(TypeUtils.isAssignable(Integer[].class, int[].class));
-        final int[] ina = null;
-        final Object[] oa;
-        // oa = ina;
         assertFalse(TypeUtils.isAssignable(int[].class, Object[].class));
-        oa = new Integer[0];
         assertTrue(TypeUtils.isAssignable(Integer[].class, Object[].class));
-        final Type bClassType = AClass.class.getField("bClass").getGenericType();
-        final Type cClassType = AClass.class.getField("cClass").getGenericType();
-        final Type dClassType = AClass.class.getField("dClass").getGenericType();
-        final Type eClassType = AClass.class.getField("eClass").getGenericType();
-        final Type fClassType = AClass.class.getField("fClass").getGenericType();
-        final AClass aClass = new AClass(new AAClass<>());
-        aClass.bClass = aClass.cClass;
+    }
+
+    @Test
+    public void testIsAssignableGenericComparableTypes() throws NoSuchFieldException {
+        final Type intComparableType = getClass().getField("intComparable").getGenericType();
+        assertTrue(TypeUtils.isAssignable(int.class, intComparableType));
+
+        final Type longComparableType = getClass().getField("longComparable").getGenericType();
+        assertFalse(TypeUtils.isAssignable(int.class, longComparableType));
+        assertFalse(TypeUtils.isAssignable(Integer.class, longComparableType));
+
+        final Type intComparableArrayType = getClass().getField("intWildcardComparable").getGenericType();
+        assertTrue(TypeUtils.isAssignable(Integer[].class, intComparableArrayType));
+    }
+
+    @Test
+    public void testIsAssignableGenericListTypes() throws SecurityException, NoSuchMethodException {
+        final Method method = getClass().getMethod("dummyMethod", List.class, List.class, List.class,
+            List.class, List.class, List.class, List.class, List[].class, List[].class,
+            List[].class, List[].class, List[].class, List[].class, List[].class);
+
+        /*
```

Comment:
It would be even more readable to pull each of these out into an appropriately named local variable and test the variables below.

e.g.

List list = types[0];
List<Object> list_object = types[1]
List<?> list_question_mark = types[2];

etc.

Up to you.

Reply:
Except use camel case for variable names, no underscores.

Reply:
I share your sympathy for having clearer type variables. It feels like the `GenericTypeHolder` class might be a better place to store generic types instead of using the `dummyMethod` field.